jax/lax/lax_control_flow.py

```diff
@@ -493,6 +495,9 @@ def scan(f, init, xs):
     xs: the value of type ``[a]`` over which to scan along the leading axis,
       where ``[a]`` can be an array or any pytree (nested Python
       tuple/list/dict) thereof with consistent leading axis sizes.
+    length: optional integer specifying the number of loop iterations, which
+      must agree with the sizes of leading axes of the arrays in ``xs`` (but can
+      be used to perform scans where no input ``xs`` are needed).
```

Comment:
Is it true that if no xs are provided then the second argument to `f` will be `None`?

Reply:
Good question! I believe that if `xs` is an empty container, then the second argument to `f` will be the same empty container. That is, if `xs=None` then the second argument to `f` is `None`, but if `xs=()` then the second argument to `f` is `()`, and if `xs=[()]` then the second argument to `f` is `[()]`.